src/Microsoft.ApplicationInsights.AspNetCore/Implementation/TelemetryConfigurationOptionsSetup.cs

```diff
@@ -119,6 +121,18 @@ private void AddTelemetryChannelAndProcessors(TelemetryConfiguration configurati
                 {
                     configuration.TelemetryProcessorChainBuilder.UseAdaptiveSampling();
                 }
+
+                // Disable heartbeat if user sets it (by default it is on)
+                if (!this.applicationInsightsServiceOptions.EnableHeartbeat)
+                {
+                    foreach (var module in TelemetryModules.Instance.Modules)
+                    {
+                        if (module is IHeartbeatPropertyManager hbeatMan)
+                        {
+                            hbeatMan.IsHeartbeatEnabled = false;
```

Comment:
Setting IsHeartbeatEnabled to false will stop the HeartBeats collected/sent by the AppServicesHeartbeatTelemetryModule, AzureInstanceMetadataTelemetryModule modules as well - right?

Reply:
Yes, this is the intention. Setting `IsHeartbeatEnabled` to false should disable heartbeats altogether.